File: QE-Selenium-Java/Activity18.java
package demos;

import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity18 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        try {
            driver.get("https://training-support.net/webelements/alerts");

            System.out.println("Page Title: " + driver.getTitle());

            WebElement simpleAlertButton = wait.until(
                    ExpectedConditions.elementToBeClickable(
                            By.xpath("//button[contains(text(),'Simple')]")));
            simpleAlertButton.click();

            Alert alert = wait.until(ExpectedConditions.alertIsPresent());

            System.out.println("Alert Text: " + alert.getText());
            alert.accept();

        } finally {
            driver.quit();
        }
    }
}

File: QE-Selenium-Python/Activity18.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

driver = webdriver.Firefox()
wait = WebDriverWait(driver, 10)

try:
    driver.get("https://training-support.net/webelements/alerts")

    print("Page Title:", driver.title)

    simple_alert_btn = wait.until(
        EC.element_to_be_clickable(
            (By.XPATH, "//button[contains(text(),'Simple')]")
        )
    )
    simple_alert_btn.click()

    alert = wait.until(EC.alert_is_present())

    print("Alert Text:", alert.text)

    alert.accept()

finally:
    driver.quit()
